pkg/auxnetdevice/auxNetDeviceProvider_test.go

```diff
@@ -0,0 +1,192 @@
+					"sys/bus/pci/devices/0000:00:00.1/foo.bar.2",
+					"sys/bus/pci/devices/0000:00:00.2",
+					"sys/bus/pci/devices/0000:00:00.3",
+					"sys/bus/pci/drivers/fake",
+				},
+				Symlinks: map[string]string{
+					"sys/bus/pci/devices/0000:00:00.1/driver": "../../../../bus/pci/drivers/fake",
+					"sys/bus/pci/devices/0000:00:00.2/driver": "../../../../bus/pci/drivers/fake",
+					"sys/bus/pci/devices/0000:00:00.3/driver": "../../../../bus/pci/drivers/fake",
+				},
+			}
+
+			defer fs.Use()()
+			utils.SetDefaultMockNetlinkProvider()
+
+			fakeSriovnetProvider := mocks.SriovnetProvider{}
+			fakeSriovnetProvider.
+				On("GetUplinkRepresentorFromAux", mock.AnythingOfType("string")).Return("net0", nil).
+				On("GetPfPciFromAux", mock.AnythingOfType("string")).Return("0000:00:00.1", nil).
+				On("GetSfIndexByAuxDev", mock.AnythingOfType("string")).Return(-1, nil).
+				On("GetNetDevicesFromAux", mock.AnythingOfType("string")).Return([]string{""}, nil)
+			utils.SetSriovnetProviderInst(&fakeSriovnetProvider)
+
+			rf := factory.NewResourceFactory("fake", "fake", true)
+			p := auxnetdevice.NewAuxNetDeviceProvider(rf)
+			config := &types.ResourceConfig{
+				DeviceType: types.AuxNetDeviceType,
+				SelectorObj: types.AuxNetDeviceSelectors{
+					DeviceSelectors: types.DeviceSelectors{},
+				},
+			}
+
+			devsToAdd := []*ghw.PCIDevice{
+				{
+					Address: "0000:00:00.1",
+					Class:   &pcidb.Class{ID: "1024"},
+					Vendor:  &pcidb.Vendor{Name: "ACompanyMakingEverything"},
+					Product: &pcidb.Product{Name: "BrandNewProduct"},
+				},
+				{
+					Address: "0000:00:00.2",
+					Class:   &pcidb.Class{ID: "1024"},
+					Vendor:  &pcidb.Vendor{Name: "ACompanyMakingEverything"},
+					Product: &pcidb.Product{Name: "BrandNewProduct"},
+				},
+				{
+					Address: "0000:00:00.3",
+					Class:   &pcidb.Class{ID: "beef"},
+					Vendor:  &pcidb.Vendor{Name: "ACompanyMakingEverything"},
+					Product: &pcidb.Product{Name: "NotWeAreLookingFor"},
+				},
+			}
+
+			err := p.AddTargetDevices(devsToAdd, 0x1024)
+			dDevs := p.GetDiscoveredDevices()
+			devs := p.GetDevices(config)
+			It("shouldn't return an error", func() {
+				Expect(err).NotTo(HaveOccurred())
+			})
+			It("should return only 1 device on GetDiscoveredDevices()", func() {
+				Expect(dDevs).To(HaveLen(2))
+			})
+			It("should return 3 auxiliary devices on GetDevices()", func() {
+				Expect(devs).To(HaveLen(3))
+			})
+		})
+	})
+	Describe("getting Filtered devices", func() {
+		Context("using selectors", func() {
+			It("should correctly filter devices", func() {
+				rf := factory.NewResourceFactory("fake", "fake", false)
+				p := auxnetdevice.NewAuxNetDeviceProvider(rf)
+				all := make([]types.HostDevice, 5)
+				mocked := make([]tmocks.AuxNetDevice, 5)
+
+				ve := []string{"8086", "8086", "1111", "2222", "3333"}
+				de := []string{"abcd", "123a", "abcd", "2222", "1024"}
+				md := []string{"tata", "toto", "toto", "toto", "titi"}
```

Comment:
can we have examples here (throughout this file) based on real data where applicable ?

Reply:
Ok